sos/plugins/podman.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (C) 2018 Red Hat, Inc. Daniel Walsh <dwalsh@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.plugins import Plugin, RedHatPlugin, UbuntuPlugin
+
+
+class Podman(Plugin):
+
+    """Podman containers
+    """
+
+    plugin_name = 'podman'
+    profiles = ('container',)
+
+    option_list = [
+        ("all", "enable capture for all containers, even containers "
+            "that have terminated", 'fast', False),
+        ("logs", "capture logs for running containers",
+            'fast', False),
+        ("size", "capture image sizes for podman ps", 'slow', False)
+    ]
+
+    def setup(self):
+        self.add_copy_spec([
+            "/etc/containers/registries.conf",
+            "/etc/containers/storage.conf",
+            "/etc/containers/mounts.conf",
+            "/etc/containers/policy.json",
+        ])
+
+        subcmds = [
+            'info',
+            'images',
+            'pod ps',
+            'pod ps -a',
+            'ps',
+            'ps -a',
+            'stats --no-stream',
+            'version',
+        ]
+
+        for subcmd in subcmds:
+            self.add_cmd_output("podman %s" % subcmd)
+
+        # separately grab ps -s as this can take a *very* long time
+        if self.get_option('size'):
+            self.add_cmd_output('podman ps -as')
+
+        self.add_journal(units="podman")
+        self.add_cmd_output("ls -alhR /etc/cni")
+
+        ps_cmd = 'podman ps -q'
+        if self.get_option('all'):
+            ps_cmd = "%s -a" % ps_cmd
+
+        img_cmd = 'podman images -q'
+        insp = set()
+
+        for icmd in [ps_cmd, img_cmd]:
+            result = self.get_command_output(icmd)
+            if result['status'] == 0:
+                for con in result['output'].splitlines():
+                    insp.add(con)
+
+        insp = list(insp)
+        if insp:
+            for container in insp:
+                self.add_cmd_output("podman inspect %s" % container)
+
+            if self.get_option('logs'):
+                for container in insp:
+                    self.add_cmd_output("podman logs -t %s" % container)
+
+
+class RedHatPodman(Podman, RedHatPlugin):
+
+    packages = ('podman')
+
+    def setup(self):
+        super(RedHatPodman, self).setup()
+
+        self.add_copy_spec([
```

Comment:
No need to wrap one path in a list - the method also accepts a single string (also from `docker`).